src/frontend/EdFi.DataManagementService.Frontend.AspNetCore/AspNetCoreFrontend.cs

```diff
@@ -33,9 +34,9 @@ public static class AspNetCoreFrontend
     /// <summary>
     /// Takes an HttpRequest and returns a unique trace identifier
     /// </summary>
-    private static string ExtractTraceIdFrom(HttpRequest request)
+    private static TraceId ExtractTraceIdFrom(HttpRequest request)
     {
-        return request.HttpContext.TraceIdentifier;
+        return new TraceId(request.HttpContext.TraceIdentifier);
```

Comment:
Concrete `TraceId` available externally because it's almost always coming from outside Core.